lib/parallel_cucumber/worker.rb

```diff
@@ -153,12 +154,43 @@ def run_batch(env, queue_tracker, results, running_total, tests)
 
     def precheck(env)
       return 'default no-op pre_check' unless @pre_check
-      continue = Helper::Command.exec_command(
-        env, 'precheck', @pre_check, @logger, @log_decoration, timeout: @batch_timeout, capture: true
-      )
-      return continue if continue
-      @logger.error('Pre-check failed: quitting immediately')
-      raise 'Pre-check failed: quitting immediately'
+
+      begin
+        return Helper::Command.exec_command(
+          env, 'precheck', @pre_check, @logger, @log_decoration, timeout: @batch_timeout, capture: true
+        )
+      rescue
+        @logger.error('Pre-check failed: quitting immediately')
+        raise 'Pre-check failed: quitting immediately'
+      end
+    end
+
+    def on_batch_error(batch_env, batch_id, error_file, tests, error)
+      unless @on_batch_error
+        return 'default no-op on_batch_error'
+      end
+
+      begin
+        error_info = {
+            'class'     => error.class,
+            'message'   => error.message,
+            'backtrace' => error.backtrace
+        }
+        batch_error_info = {
+            'batch_id' => batch_id,
+            'tests'    => tests,
+            'error'    => error_info
+        }.to_json
+        File.write(error_file, batch_error_info)
+        command = "#{@on_batch_error} #{error_file}"
+        return Helper::Command.exec_command(
+            batch_env, 'on_batch_error', command, @logger, @log_decoration, timeout: @batch_timeout, capture: true
+        )
+      rescue
+        message = 'on-batch-error failed: quitting immediately'
+        @logger.error(message)
+        raise message
```

Comment:
Print error message and do not fail worker.

Reply:
👍